test/noinline_tests.jl

```diff
@@ -0,0 +1,40 @@
+@testset "Noinline" begin
+    using ReLint
+
+    source = raw"""
+        function with_errors()
+            @noinline g(x.a)
+            @noinline g("$(x)")
+            @noinline h(g(x))
+            @noinline i(1 + 2)
+            @noinline h(x) + 1
+            @noinline @eval 1 + 1
+            @noinline g(x[1])
+            @noinline g(x[:a])
+            @noinline (1:10) .+ 4
+        end
+
+        function without_errors()
+            @noinline g(x)
+            @noinline g(1, 2, 3, "abc", 'a')
+            @noinline 1 + 1
+            @noinline x[1]
+        end
```

Comment:
@bergel could this be updated this to be aware of splatting?

Currently this triggers if there’s splatting in the call e.g. these all cause a lint violation but should not:
```
@noinline foo(x...)
@noinline foo(x, y...)
@noinline foo(x; kws...)
```